internal/collections/named.go

```diff
@@ -122,6 +122,7 @@ func (c *NamedCollectionNames) FindAll() []types.MatchData {
 		res = append(res, &corazarules.MatchData{
 			VariableName_: c.name,
 			Variable_:     c.variable,
+			Key_:          k,
```

Comment:
It adds name redundancy inside MatchData populating with the same value also the `Key_ `, which will later be read and concatenated

Reply:
Discussion about this came up here

https://github.com/corazawaf/coraza/pull/606#discussion_r1091488393

But if there are CRS tests to verify then this change looks good